ocsp_status.go

```
@@ -31,129 +28,102 @@ func createConn(certURL string) *tls.Conn {
-		panic(err.Error())
-	}
+	ocspResp := tlsConn.OCSPResponse()
+	tlsConn.Close()
 
-	if dir != "" {
-		err := ioutil.WriteFile(dir, ocspResp, 0644)
-		if err != nil {
-			panic("Error writing to file: " + err.Error())
+	if ocspResp == nil {
+		fmt.Println("No OCSP response stapled")
+
+		reqMethod := http.MethodPost
+		if get {
+			reqMethod = http.MethodGet
 		}
-	}
 
-	parseOCSPResp(ocspResp, issuerCert)
+		ocspReq := ocsptools.CreateOCSPReq(ocspURL, rootCert, issuerCert, reqMethod, hash)
+		ocspResp = ocsptools.GetOCSPResponse(ocspReq, dir)
+
+		return ocsptools.ParseAndLint(ocspResp, issuerCert)
+	} else {
+		fmt.Println("Stapled OCSP Response")
+		return ocsptools.ParseAndLint(ocspResp, issuerCert)
+	}
 }
 
+// main parses the users commandline arguments & flags and then runs the appropriate functions
 func main() {
-	print := flag.Bool("print", false, "Whether to print certificate or not")
-	dir := flag.String("dir", "", "Where to write OCSP response, if blank don't write")
-	reqType := flag.String("type", http.MethodPost, "Whether to use GET or POST for OCSP request")
-	in := flag.Bool("in", false, "Whether to read in an OCSP Response or not")
+	// TODO: extract flag descriptions into constants?
+	inresp := flag.Bool("inresp", false, "Whether to read in an OCSP responses or not")
+	incert := flag.Bool("incert", false, "Whether to read in certificate files or not")
+	ocspurl := flag.String("ocspurl", "", "User provided OCSP url, default fetch from certificate")
+	print := flag.Bool("print", false, "Whether to print certificate or not") // may remove this print flag
+	get := flag.Bool("get", false, "Whether to use GET for OCSP request")
+	dir := flag.String("dir", "", "Where to write OCSP response, if blank don't write")	
 
 	flag.Parse()
 
-	if *in {
+	if *inresp {
 		respFiles := flag.Args()
-
 		for _, respFile := range respFiles {
-			ocspResp, err := ioutil.ReadFile(respFile)
+			lintFile(respFile)
+		}
+	} else if *incert {
```

Maybe document somewhere that if inresp is true, then the value of incert no longer matters?

---

This is definitely something I've been thinking about. The code should actually be written in a way so that it is not possible to have both the `-inresp` and `-incert` flags at the same time.

---

I agree. I was thinking a boolean similar to how you did for get vs post, but it looks like there are more than 2 options. Maybe raise an error if both are set to prevent that case from appearing as they start using the tool.